website/content/v1.10/kubernetes-guides/configuration/inlinemanifests.md

```diff
@@ -0,0 +1,176 @@
+---
+title: "inlineManifests and extraManifests"
+description: "Learn what inlineManifests and extraManifests are, how they differ, and why they matter."
+---
+
+`inlineManifests` and `extraManifests` allow you to automatically apply Kubernetes resources to your cluster during installation and upgrades.
+
+Both are designed to automate the provisioning of components like CNIs and other static infrastructure, but they differ in how the manifest content is sourced and applied.
+They are not meant for deploying applications or frequently changing services.
+For those, it's better to use a GitOps or CI/CD tool.
+
+## inlineManifests
+
+`inlineManifests` are defined directly within the machine configuration file.
+The YAML content is embedded inside the `inlineManifests` section, making it ideal for tightly coupled resources that need to be provisioned as soon as the node boots up.
+
+Here’s an example of how to configure a cluster using an `inlineManifest`:
+
+```yaml
+cluster:
+  inlineManifests:
+    - name: my-app
+      contents: |
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          name: my-application
+        spec:
+          # ... deployment specification
+```
+
+## extraManifests
+
+`extraManifests` are Kubernetes manifests fetched from external, unauthenticated HTTP sources such as GitHub, raw file servers, or gists.
+
+You define them in the `extraManifests` section of the machine configuration.
+They’re best suited for shared, versioned, or centrally managed resources.
+
+These manifests are pulled directly by the node during configuration.
+If the node doesn’t have network access to the HTTP endpoint hosting the manifest the installation will fail.
+
+Similarly, if the endpoint is down or returns an error, the manifest will not be applied, and the machine configuration will fail as a result.
+
+Here’s how to reference `extraManifests`:
+
+```yaml
+cluster:
+  extraManifests:
+    - "https://raw.githubusercontent.com/example/repo/main/manifest.yaml"
+    - "https://gist.githubusercontent.com/user/gist-id/raw/manifest.yaml"
+```
+
+## Resource Ordering Considerations
+
+You must note that although `inlineManifests` and `extraManifests` are defined as arrays, Talos does not guarantee that the items will be applied in the order they appear.
+
+If your resources depend on one another and must be applied in a specific order, define them in a single manifest entry and separate each resource using the `---` YAML document separator.
```

Comment:
this whole section is not how Talos works I have no idea where it came from.

Talos sorts manifests in the following order:

* namespaces
* CRDs
* everything else in alphabetical order of `name:` property

Kubernetes doesn't technically require a specific order of manifests to apply, so it should be fine.

Plus even if one manifest fails to apply before another one is applied, Talos anyways retries the whole process, so it will eventually apply.

Reply:
it doesn't really matter where the manifests came from - Talos built-in, inline, extra - they are all processed in the end by extracting all resources and then applying resource-by-resource in the order above

Reply:
We'll make sure that info gets added to the docs

Reply:
I got it from a comment in the previous round of reviews. I will delete